action/clihelper_test.go

```diff
@@ -0,0 +1,41 @@
+package action
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_askForGitConfigUser(t *testing.T) {
+	s := New("test-init")
+	s.isTerm = true
+
+	_, _, err := s.askForGitConfigUser()
+	if err == nil {
+		t.Error("Did not return error")
+	}
+}
+
+func Test_askForGitConfigUserNonInteractive(t *testing.T) {
+	s := New("test-init")
+	// for explicitness
+	s.isTerm = false
+
+	keyList, err := s.gpg.ListPrivateKeys()
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	name, email, _ := s.askForGitConfigUser()
+
+	// tests cannot know whether keyList returned empty or not.
+	// a better distinction would require mocking/patching
+	// calls to s.gpg.ListPrivateKeys()
```

Comment:
Yes, that's probably too much overhead here. That's why we made part of the tests as integration tests.